File: code/generate_strategies/RA(name)/python/python/function_boltons_cacheutils_LRU.py
from boltons.cacheutils import LRU


class LRUCacheManager:
    def manage_cache_operations(self, cache_size, operations_str):
        cache = LRU(cache_size)
        results = []
        operations = operations_str.split(',')
        if operations_str == '':
            return ''
        i = 0
        while i < len(operations):
            operation_type = operations[i]
            key = operations[i + 1]

            if operation_type == "set":
                value = operations[i + 2]
                cache[key] = value
                i += 3
            elif operation_type == "get":
                result = cache.get(key)
                results.append(str(result) if result is not None else "None")
                i += 2
            else:
                raise ValueError("Invalid operation type")

        return ",".join(results)

File: code/generate_strategies/Direct/java/function_boltons_cacheutils_LRU.java
package com.example;

import com.google.common.cache.CacheBuilder;
import com.google.common.cache.Cache;
import java.util.ArrayList;
import java.util.List;

public class LRUCacheManager {
    public String manage_cache_operations(int cache_size, String operations_str) {
        class LRUCacheManager {
            private final Cache<String, String> cache;

            public LRUCacheManager(int size) {
                cache = CacheBuilder.newBuilder()
                        .maximumSize(size)
                        .build();
            }

            public void set(String key, String value) {
                cache.put(key, value);
            }

            public String get(String key) {
                return cache.getIfPresent(key);
            }
        }

        if (operations_str.isEmpty()) {
            return "";
        }

        LRUCacheManager cacheManager = new LRUCacheManager(cache_size);
        List<String> results = new ArrayList<>();
        String[] operations = operations_str.split(",");

        int i = 0;
        while (i < operations.length) {
            String operation_type = operations[i];
            String key = operations[i + 1];

            if ("set".equals(operation_type)) {
                String value = operations[i + 2];
                cacheManager.set(key, value);
                i += 3;
            } else if ("get".equals(operation_type)) {
                String result = cacheManager.get(key);
                results.add(result == null ? "None" : result);
                i += 2;
            } else {
                throw new IllegalArgumentException("Invalid operation type");
            }
        }

        return String.join(",", results);
    }
}